mongomock/store.py

```diff
@@ -104,3 +107,51 @@ def __len__(self):
     def documents(self):
         for doc in six.itervalues(self._documents):
             yield doc
+
+    def remove_expired_documents(self):
```

Comment:
I would prefer if the responsibility of never returning expired document would land in the store class. Therefore I believe this method should be protected and called in the other methods in this class.

Two reasons why I prefer it to be the responsibility of the store rather than the collection:
* the store is mocking the server's storage, whereas the collection is the client's view of it and the expiration of documents is done in the server
* there is no reasons ever that the store should returned expired document

Reply:
Gotcha. I had thought of that originally, but once I saw that basically every method of the store would need to call this method, I started lurking elsewhere :wink: